tests/test_lsap.py

```diff
@@ -94,6 +94,13 @@ def test_linear_sum_assignment_input_validation():
     eye[:, 0] = np.inf
     assert_raises(ValueError, linear_sum_assignment, eye)
 
+    # Note that solve_lsap requires that num_rows cannot exceed num_cols
```

Comment:
We should check the shape of the matrix at the start and operate on the transpose if it is the wrong shape.

Reply:
linear_sum_assignment does that. solve_lsap does not. (They are two different functions in dynamic_lsap.py)